web/src/composebox_typeahead.ts

```diff
@@ -1173,6 +1172,15 @@ export function content_typeahead_selected(
             // "beginning" contains all the text before the cursor, so we use lastIndexOf to
             // avoid any other stream+topic mentions in the message.
             const syntax_start_index = beginning.lastIndexOf("#**");
+            const topic_start_index = beginning.lastIndexOf(">");
+            const topic_name = item.topic;
+            // eliminate #** while deducing stream name
```

Comment:
I think this would read a bit more clearly like this:

```js
// Get the stream name, not including "#**" at the start or ">" at the end.
const topic_start_index = beginning.lastIndexOf(">");
const stream_name = beginning.slice(syntax_start_index + 3, topic_start_index);

if (item.topic === stream_name) {
  // The user opted to select only the stream and specify a topic.
  beginning = beginning.slice(0, syntax_start_index) + "#**" + stream_name + "** ";
  break;
}
```